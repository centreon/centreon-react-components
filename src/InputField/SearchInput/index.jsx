import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Input from '@material-ui/core/Input';
import InputAdornment from '@material-ui/core/InputAdornment';
import SearchIcon from '@material-ui/icons/Search';

const useAdornmentStyles = makeStyles((theme) => ({
  searchIcon: {
    color: theme.palette.grey[400],
  },
}));

const StartAdornment = () => {
  const classes = useAdornmentStyles();

  return (
    <InputAdornment position="start">
      <SearchIcon
        classes={{
          root: classes.searchIcon,
        }}
      />
    </InputAdornment>
  );
};

const useStyles = makeStyles((theme) => {
  // took it from material-ui source code
  const light = theme.palette.type === 'light';
  const bottomLineColor = light
    ? 'rgba(0, 0, 0, 0.42)'
    : 'rgba(255, 255, 255, 0.7)';

  return {
    root: {
      backgroundColor: theme.palette.common.white,
      border: `1px solid ${theme.palette.grey[400]}`,
      height: 32,
      paddingLeft: 6,
      fontSize: 13,
    },
    input: {
      paddingBottom: 4,
    },
    underline: {
      '&:hover:not($disabled):before': {
        borderBottom: `2px solid ${bottomLineColor}`,
      },
    },
    disabled: {},
  };
});

const SearchInput = forwardRef((props, ref) => {
  const classes = useStyles();

  return (
    <Input
      classes={{
        root: classes.root,
        input: classes.input,
        underline: classes.underline,
      }}
      margin="dense"
      ref={ref}
      startAdornment={<StartAdornment />}
      {...props}
    />
  );
});

SearchInput.propTypes = {
  placeholder: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

SearchInput.defaultProps = {
  placeholder: 'Search',
  value: null,
};

export default SearchInput;
