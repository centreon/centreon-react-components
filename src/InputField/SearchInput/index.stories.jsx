import React from 'react';

import SearchInput from '.';

export default { title: 'InputField/Search' };

export const normal = () => <SearchInput />;
