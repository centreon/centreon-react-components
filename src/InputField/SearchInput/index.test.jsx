import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SearchInput from '.';

describe('SearchInput', () => {
  it('renders correctly', () => {
    const { container } = render(
      <SearchInput placeholder="search" onChange={jest.fn()} />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('tiggers change', () => {
    const mockOnChange = jest.fn();

    const { getByPlaceholderText } = render(
      <SearchInput onChange={mockOnChange} />,
    );

    const input = getByPlaceholderText('Search');
    fireEvent.change(input, { target: { value: 'my search' } });

    expect(mockOnChange).toBeCalled();
  });
});
