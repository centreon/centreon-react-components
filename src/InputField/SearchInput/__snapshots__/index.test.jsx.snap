// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders correctly 1`] = `
<div
  class="MuiInputBase-root MuiInput-root makeStyles-root-1 MuiInput-underline makeStyles-underline-3 MuiInputBase-adornedStart MuiInputBase-marginDense MuiInput-marginDense"
>
  <div
    class="MuiInputAdornment-root MuiInputAdornment-positionStart"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-searchIcon-39"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
  </div>
  <input
    class="MuiInputBase-input MuiInput-input makeStyles-input-2 MuiInputBase-inputAdornedStart MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
    placeholder="search"
    type="text"
    value=""
  />
</div>
`;
